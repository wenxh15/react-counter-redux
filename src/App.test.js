import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

import ConnectedApp, { mapStateToProps, App } from "./App";

describe("<AppÂ />", () => {
  const actions = {
    decreaseCounter: jest.fn(),
    addCounter: jest.fn()
  };

  const mockStore = configureStore();

  const initialState = {
    count: 0
  };

  const props = {
    actions,
    ...mapStateToProps(initialState)
  };

  const store = mockStore(initialState);

  it("renders connected component", () => {
    const component = renderer.create(
      <Provider store={store}>
        <ConnectedApp />
      </Provider>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("counter button interact when onOclick", () => {
    const component = renderer.create(<App {...props} />);
    const addButton = component.root.findByProps({
      className: "+"
    });

    addButton.props.onClick();

    expect(actions.addCounter).toHaveBeenCalledTimes(1);

    const decreaseButton = component.root.findByProps({
      className: "-"
    });
    decreaseButton.props.onClick();
    expect(actions.decreaseCounter).toHaveBeenCalledTimes(1);
  });
});
