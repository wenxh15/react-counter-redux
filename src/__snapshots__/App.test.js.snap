// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ServiceRequestsSummaryÂ /> renders connected component 1`] = `
<div
  className="App"
>
  <h1>
    Counter
  </h1>
  <button
    className="-"
    onClick={[Function]}
  >
    -
  </button>
  <p>
    0
  </p>
  <button
    className="+"
    onClick={[Function]}
  >
    +
  </button>
</div>
`;
